diff --git a/src/packages/tests/src/__tests__/integration/mariadb/__database.ts b/src/packages/tests/src/__tests__/integration/mariadb/__database.ts
index f143c928..44160832 100644
--- a/src/packages/tests/src/__tests__/integration/mariadb/__database.ts
+++ b/src/packages/tests/src/__tests__/integration/mariadb/__database.ts
@@ -19,11 +19,16 @@ export const database = {
     })
   },
   beforeEach: async (db, sqlScenario, ctx) => {
-    const sqlUp = `
-    DROP DATABASE IF EXISTS ${ctx.id};
-    CREATE DATABASE ${ctx.id};
-    USE ${ctx.id};`
-    await db.query(sqlUp + sqlScenario)
+    await db.query("USE " + process.env.VT_DATABASE);
+    const foreign_keys = await db.query({rowsAsArray: true, sql: "SELECT DISTINCT TABLE_NAME, CONSTRAINT_NAME FROM information_schema.KEY_COLUMN_USAGE WHERE REFERENCED_TABLE_NAME IS NOT NULL"});
+    for(let row of foreign_keys) {
+      await db.query("ALTER TABLE `" + row[0] + "` DROP FOREIGN KEY `" + row[1] + "`");
+    }
+    const tables = await db.query({rowsAsArray: true, sql: "SHOW TABLES"});
+    for(let row of tables) {
+      await db.query("DROP TABLE `" + row[0] + "`");
+    }
+    await db.query(sqlScenario)
   },
   close: (db) => db.end(),
 } as Input<mariadb.Connection>['database']
@@ -31,7 +36,8 @@ export const database = {
 function getConnectionInfo(ctx: Context) {
   const serviceConnectionString =
     process.env.TEST_MARIADB_BASE_URI || 'mysql://root:root@localhost:4306'
-  const connectionString = `${serviceConnectionString}/${ctx.id}`
+  const connectionString = process.env.TEST_MARIADB_URI || `${serviceConnectionString}/${ctx.id}`
+  console.log("connectionString = ", connectionString);
   const credentials = uriToCredentials(connectionString)
 
   return {
diff --git a/src/packages/tests/src/__tests__/integration/mysql/__database.ts b/src/packages/tests/src/__tests__/integration/mysql/__database.ts
index 2a1d7a85..ce580fc6 100644
--- a/src/packages/tests/src/__tests__/integration/mysql/__database.ts
+++ b/src/packages/tests/src/__tests__/integration/mysql/__database.ts
@@ -18,11 +18,16 @@ export const database = {
     })
   },
   beforeEach: async (db, sqlScenario, ctx) => {
-    const sqlUp = `
-    DROP DATABASE IF EXISTS ${ctx.id};
-    CREATE DATABASE ${ctx.id};
-    USE ${ctx.id};`
-    await db.query(sqlUp + sqlScenario)
+    await db.query("USE " + process.env.VT_DATABASE);
+    const foreign_keys = await db.query({rowsAsArray: true, sql: "SELECT DISTINCT TABLE_NAME, CONSTRAINT_NAME FROM information_schema.KEY_COLUMN_USAGE WHERE REFERENCED_TABLE_NAME IS NOT NULL"});
+    for(let row of foreign_keys) {
+      await db.query("ALTER TABLE `" + row[0] + "` DROP FOREIGN KEY `" + row[1] + "`");
+    }
+    const tables = await db.query({rowsAsArray: true, sql: "SHOW TABLES"});
+    for(let row of tables) {
+      await db.query("DROP TABLE `" + row[0] + "`");
+    }
+    await db.query(sqlScenario)
   },
   close: (db) => db.end(),
 } as Input<mariadb.Connection>['database']
@@ -30,7 +35,8 @@ export const database = {
 function getConnectionInfo(ctx: Context) {
   const serviceConnectionString =
     process.env.TEST_MYSQL_BASE_URI || 'mysql://root:root@localhost:3306'
-  const connectionString = `${serviceConnectionString}/${ctx.id}`
+  const connectionString = process.env.TEST_MYSQL_URI || `${serviceConnectionString}/${ctx.id}`
+  console.log("connectionString = ", connectionString);
   const credentials = uriToCredentials(connectionString)
 
   return {
